src/litdata/streaming/downloader.py

```diff
@@ -69,12 +73,28 @@ def __init__(
         self, remote_dir: str, cache_dir: str, chunks: List[Dict[str, Any]], storage_options: Optional[Dict] = {}
     ):
         super().__init__(remote_dir, cache_dir, chunks, storage_options)
+        if _USE_EXPERIMENTAL_RUST:
+            from litdata._core import S3ByteRangeDownloader
+
+            self.s3_byte_range_downloader = S3ByteRangeDownloader()
+
         self._s5cmd_available = os.system("s5cmd > /dev/null 2>&1") == 0
 
         if not self._s5cmd_available or _DISABLE_S5CMD:
             self._client = S3Client(storage_options=self._storage_options)
 
     def download_file(self, remote_filepath: str, local_filepath: str, remote_chunk_filename: str = "") -> None:
+        if _USE_EXPERIMENTAL_RUST and not remote_filepath.endswith(_INDEX_FILENAME):
+            if remote_chunk_filename not in self.chunk_filename_to_chunk_bytes:
+                raise ValueError(f"Chunk filename {remote_chunk_filename} not found in chunk_filename_to_chunk_bytes")
+            self.s3_byte_range_downloader.byte_range_download(
+                remote_filepath,
+                local_filepath,
+                num_of_bytes=self.chunk_filename_to_chunk_bytes[remote_chunk_filename],
+                num_workers=128,
+            )
```

Comment:
Here, I have a few suggestions for the parameters:  

Instead of calculating the chunk byte size based on the number of workers, I believe it would be better to pass the byte range size directly and perhaps include another parameter, such as `max_connections` to control the concurrent connections.
> That we will have better control over byte range chunk size

Reply:
Additionally, during the benchmark test, as we observed : memory usage exceeded the 70GB range. In comparison, memory usage for s5cmd and boto3 remained at or below 30GB.